tests/integration_tests/performance/test_snapshot_restore_performance.py

```diff
@@ -214,164 +195,135 @@ def consume_output(cons, result):
-def test_snap_restore_performance(bin_cloner_path, results_file_dumper):
-    """
-    Test the performance of snapshot restore.
-
-    @type: performance
-    """
-    logger = logging.getLogger(TEST_ID)
-    artifacts = ArtifactCollection(_test_images_s3_bucket())
-    microvm_artifacts = ArtifactSet(artifacts.microvms(keyword="2vcpu_1024mb"))
-    kernel_artifacts = ArtifactSet(artifacts.kernels())
-    disk_artifacts = ArtifactSet(artifacts.disks(keyword="ubuntu"))
-
-    logger.info("Testing on processor %s", get_cpu_model_name())
-
-    # Create a test context and add builder, logger, network.
-    test_context = TestContext()
-    test_context.custom = {
-        "builder": MicrovmBuilder(bin_cloner_path),
-        "logger": logger,
-        "name": TEST_ID,
-        "results_file_dumper": results_file_dumper,
-        "workload": "restore",
-    }
-
-    test_matrix = TestMatrix(
-        context=test_context,
-        artifact_sets=[microvm_artifacts, kernel_artifacts, disk_artifacts],
-    )
-    test_matrix.run_test(snapshot_workload)
-
-
-def snapshot_scaling_vcpus(context, st_core, vcpu_count=10):
+@pytest.mark.parametrize("vcpu_count", [1, 2, 4, 8, 10])
+def test_snapshot_scaling_vcpus(
+    bin_cloner_path, microvm_factory, rootfs, guest_kernel, vcpu_count, st_core
+):
     """Restore snapshots with variable vcpu count."""
-    workload = context.custom["workload"]
-    for i in range(vcpu_count):
-        env_id = (
-            f"{context.kernel.name()}/{context.disk.name()}/"
-            f"{BASE_VCPU_COUNT + i}vcpu_{BASE_MEM_SIZE_MIB}mb"
-        )
-
-        st_prod = st.producer.LambdaProducer(
-            func=get_snap_restore_latency,
-            func_kwargs={
-                "context": context,
-                "vcpus": BASE_VCPU_COUNT + i,
-                "mem_size": BASE_MEM_SIZE_MIB,
-            },
-        )
-        st_cons = default_lambda_consumer(env_id, workload)
-        st_core.add_pipe(st_prod, st_cons, f"{env_id}/{workload}")
+    guest_config = f"{vcpu_count}vcpu_{BASE_MEM_SIZE_MIB}mb"
+    env_id = f"{guest_kernel.name()}/{rootfs.name()}/{guest_config}"
+    st_prod = st.producer.LambdaProducer(
+        func=get_snap_restore_latency,
+        func_kwargs={
+            "vm_builder": MicrovmBuilder(bin_cloner_path),
+            "microvm_factory": microvm_factory,
+            "guest_kernel": guest_kernel,
+            "rootfs": rootfs,
+            "vcpus": vcpu_count,
+            "mem_size": BASE_MEM_SIZE_MIB,
+        },
+    )
+    st_cons = default_lambda_consumer(env_id, WORKLOAD)
+    st_core.add_pipe(st_prod, st_cons, f"{env_id}/{WORKLOAD}")
+    st_core.name = TEST_ID
+    st_core.custom["guest_config"] = guest_config
+    st_core.run_exercise()
 
 
-def snapshot_scaling_mem(context, st_core, mem_exponent=9):
+@pytest.mark.nonci
+# exponent=8 takes around 400s seconds
+@pytest.mark.timeout(10 * 60)
+@pytest.mark.parametrize("mem_exponent", range(1, 9))
```

Comment:
```suggestion
@pytest.mark.timeout(10 * 60)
# `mem_exponent=8` takes ~400s seconds
@pytest.mark.parametrize("mem_exponent", range(1, 9))
```

Reply:
Black did not complain, but it may be easier to read.